### PR TITLE
Add initial webkit theming

### PR DESCRIPTION
**Pull request by Foldex**
Should serve as decent starting point, and a marked improvement over Steam's endless sea of blue gradients.

### CODE REVIEW

File: Adwaita/resource/webkit.css [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+*/
+
+/* -------------- */
+/* --- Colors --- */
+/* -------------- */
+
+:root {
+    --bg: #242424;
+    --fg: #ffffff;
+    --radius: 15px;
```
Comment (tkashkin) [inline Adwaita/resource/webkit.css:18]: [libadwaita stylesheet](https://gitlab.gnome.org/GNOME/libadwaita/-/tree/main/src/stylesheet) uses `6px` radius for small widgets like buttons and `12px` for larger widgets like cards, rows, windows.

From https://gitlab.gnome.org/GNOME/libadwaita/-/blob/main/src/stylesheet/_common.scss:
```scss
$button_radius: 6px;
$card_radius: $button_radius + 6;
$menu_radius: 6px;
$window_radius: $button_radius + 6;
$popover_radius: $window_radius;
```

**Comment by tkashkin** [comment]
I don't think the skin should change anything other than colors and scrollbars by default, at least before the `webkit.css` is polished enough. Making it an optional patch would be good.

Do you know if Steam can load separate CSS files with `@import` rules? Should be much easier to make some parts of the stylesheet optional if it works.

**Comment by Foldex** [comment]
>I don't think the skin should change anything other than colors and scrollbars by default, at least before the webkit.css is polished enough. Making it an optional patch would be good.

Fair enough, it's only a vaguely adwaita-ish WIP as I'm being lazy with the details for now. I'll have to do some digging through libadwaita with a finer tooth comb later.
Is colors and scrollbars a hardline stance for the minimal version, or can I wiggle in one or two elements that otherwise would look out of place?

>Do you know if Steam can load separate CSS files with @import rules? Should be much easier to make some parts of the stylesheet optional if it works.

Was the first thing I tried, sadly it seems unsupported as far as I could tell. We could still split it into separate files and make the installer script concat combinations together for different versions though. If not that, we can always fall back onto the big ugly patch plan.

File: Adwaita/resource/webkit.css (hunk 1 above)
Comment (Foldex) [reply Adwaita/resource/webkit.css]: Should be good now https://github.com/tkashkin/Adwaita-for-Steam/pull/6/commits/a698e9ff151fc4c95ad5f4238b1a146795823ed4

**Comment by tkashkin** [comment]
> Is colors and scrollbars a hardline stance for the minimal version, or can I wiggle in one or two elements that otherwise would look out of place?

Not really a hard requirement, I just don't want the skin to change too much of the library UI initially, at least not by default. Bigger changes can be added over time and enabled by default after some testing and polishing.

**Comment by tkashkin** [comment]
Yes, I'm fine with that. Let me know when the base part is ready for review.

---

Some code style preferences:
* Indent with tabs instead of spaces
* Put brackets on a separate line

```css
selector1,
selector2
{
	property: value;
}
```

**Comment by pinkuuu1** [comment]
Looks awesome so far! Is there plans to fix the friends list page with that blue gradient on the top?

Thank you both for your hard work!! It is very appritiated :)

**Comment by Foldex** [comment]
> Looks awesome so far! Is there plans to fix the friends list page with that blue gradient on the top?

The base version of the theme doesn't touch chat, but the latest full version should have the chat pretty much entirely themed.
Do you have a screenshot of that?

> Thank you both for your hard work!! It is very appritiated :)

:+1: 


@tkashkin

The base theme should be good enough for a review at this point.
Pretty much only basic colors for the Library/Collections/Downloads page, and scrollbars.

The full theme implements Chat/Game Details view fairly well, but they're not included in base for now. 
The more rounded aesthetic and basic Store/Community background colors are here as well, but I'm not too interested in theming every aspect of those pages.

**Comment by tkashkin** [comment]
@Foldex 

The base theme looks good, so far I've noticed a few small issues:

1. Home and Collections buttons look a bit weird without a background:
image

2. Some of the search bar corners are not rounded.
3. Search bar has no background when filters popup is visible:
image

4. Full game name tooltip is blue when the game is not selected and transparent when it is selected:
image
image

5. Library scrollbar background is slightly blue, can be hard to notice:
image

6. What's New shelf has a darker background in the default skin and it seems to look better (or maybe I'm just used to its default style). It will probably look better with `--view_bg_color` as a background and maybe with a bottom border, see the screenshots view in GNOME Software for example:
image

---

I think the sidebar buttons (Home, Collections), filter dropdown and search field can be rounded to 6px in the base theme, it's a small enough change that should make the sidebar look better.

**Comment by Foldex** [comment]
1-6 should be fixed now.

**Comment by pinkuuu1** [comment]
image
Here is a screenshot of that friends pannel thing btw, not sure if it is a feature or not.

Cheers!

**Comment by Foldex** [comment]
It did stand out and was simple enough to fix, but I'll be upfront and say support for the web portions that just load Steam's website is going to be minimal from me.

Those areas are much larger in scope compared to the core areas of the client, and things like the Store are much more likely to undergo changes for special events and the like, leading to a higher burden of maintenance.

Aside from very glaring issues, it's mostly just a background change to somewhat fit in and is probably staying behind the `full` flag for this reason.

**Comment by tkashkin** [comment]
@Foldex 

I've tweaked game list paddings a bit to make it look more similar to libadwaita sidebars.
I've also tried to increase the list item height, but it seems to be hardcoded to 24px in JS, same issue as in VGUI menus and lists.

The only remaining issue I've noticed is when multiple games are selected:
image

I wasn't able to fix it easily without breaking other things, maybe you'll be able to do that.

Other than this the base theme seems to be ready to merge.

**Comment by tkashkin** [comment]
@Foldex 

A couple of very small and probably not even that important issues left:

1. Hover a selected game with download progress or cloud sync error when there's enough width (no overlay):
image

2. Same, but with multiple games selected and with overlay:
image

I can wait for you to try to fix this or I can merge this PR now and leave this to be fixed later.

**Comment by Foldex** [comment]
@tkashkin 
Whac-a-Mole round three?

**Comment by tkashkin** [comment]
@Foldex 
Everything seems to be fixed now, thanks.

**Pull request merged**